llvm/test/CodeGen/DirectX/legalize-memcpy.ll

```diff
@@ -0,0 +1,174 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 5
+; RUN: opt -S -dxil-legalize -dxil-finalize-linkage -mtriple=dxil-pc-shadermodel6.3-library %s | FileCheck %s
+
+@outputStrides = external local_unnamed_addr addrspace(2) global [2 x <4 x i32>], align 4
```

Comment:
Does this pass come before scalarization? I ask because the native vectors won't be present in final DXIL until 6.9 and would make an odd choice if this came after scalarization.

Reply:
According to [DirectXTargetMachine.cpp:L110-L116](https://github.com/llvm/llvm-project/blob/main/llvm/lib/Target/DirectX/DirectXTargetMachine.cpp#L110-L116) this legalize pass comes after scalarization.
The test `replace_2x4xint_global_memcpy_test` reflects that of LLVM IR I saw when compiling DML shaders, which use shader model 6.6, yet the alloca and the global variable clearly contain vector types.
If native vectors aren't supposed to be present in the final DXIL until 6.9, then I suppose something is going very wrong here?

Reply:
I filed an issue regarding these vectors appearing in the IR after data scalarization and array flattening:
https://github.com/llvm/llvm-project/issues/140143